components/director/run.sh

```diff
@@ -256,16 +264,42 @@ cat <<EOF > /tmp/keyConfig
 }
 EOF
 
-kubectl create secret generic "$CLIENT_CERT_SECRET_NAME" --from-literal="$APP_EXTERNAL_CLIENT_CERT_KEY"="$APP_EXTERNAL_CLIENT_CERT_VALUE" --from-literal="$APP_EXTERNAL_CLIENT_KEY_KEY"="$APP_EXTERNAL_CLIENT_KEY_VALUE" --save-config --dry-run=client -o yaml | kubectl apply -f -
+kubectl create secret generic "$CLIENT_CERT_SECRET_NAME" --from-literal="$APP_EXTERNAL_CLIENT_CERT_KEY"="$APP_EXTERNAL_CLIENT_CERT_VALUE" --from-literal="$APP_EXTERNAL_CLIENT_KEY_KEY"="$APP_EXTERNAL_CLIENT_KEY_VALUE" --save-config --dry-run -o yaml | kubectl apply -f -
```

Comment:
From kubectl v1.18+ `--dry-run` without any value has been deprecated. (https://github.com/kubernetes/kubernetes/pull/105327)

k8s docs say that explicit value must be set: `Must be "none", "server", or "client"` (https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands)

If someone uses an older kubectl version locally `--dry-run` may work, but given the fact that before this PR the script used `--dry-run=client` I guess most of the people adjusted their local kubectl version and now when the `=client` value is removed the script won't work. (I use v1.23.0 locally and it failed with: `error: --dry-run flag without a value was specified. A value must be set: "none", "server", or "client".`)

So, I think it would be better to leave `--dry-run=client`.

Reply:
I double-check and see that kubectl version that was used in the prow image is 1.21: https://github.com/kyma-project/test-infra/blob/main/prow/images/bootstrap/Dockerfile#L58

Will revert this change.